pkg/engine/update.go

```diff
@@ -95,28 +95,55 @@ func Update(u UpdateInfo, ctx *Context, opts UpdateOptions, dryRun bool) (Resour
 }
 
 func newUpdateSource(
-	opts planOptions, proj *workspace.Project, pwd, main string,
+	client deploy.BackendClient, opts planOptions, proj *workspace.Project, pwd, main string,
 	target *deploy.Target, plugctx *plugin.Context, dryRun bool) (deploy.Source, error) {
 
-	// Figure out which plugins to load by inspecting the program contents.
-	plugins, err := plugctx.Host.GetRequiredPlugins(plugin.ProgInfo{
+	// Before launching the source, ensure that we have all of the plugins that we need in order to proceed.
+	//
+	// There are two places that we need to look for plugins:
+	//   1. The language host, which reports to us the set of plugins that the program that's about to execute
```

Comment:
How much value does this add? Given that this is advisory--and only implemented by NodeJS--should we just rely on these to be installed already? Not sure that it's worth further bifurcating the language experience.

Reply:
I dunno, I think it's of marginal value but it's here in this PR because this is what this code path used to do. I think what we actually want is @lukehoban's suggestion of sending a version alongside `registerResource` and, once we can do that, we basically don't need to do this at all anymore.